rasa_core/trackers.py

```diff
@@ -113,6 +114,13 @@ def current_state(self,
             "events": evts
         }
 
+    def past_states(self, domain):
+        # type: (Domain) -> deque
+        """Generate the past states of this tracker based on the history."""
+
+        generated_states = domain.states_for_tracker_history(self)
+        return deque((frozenset(s.items()) for s in generated_states))
```

Comment:
I'm confused a bit now, will it cache the probabilities as well?

Reply:
yes it will cache tuples of the name and the value, so `frozenset([(u'intent_greet', 1.0), (u'prev_action_listen', 1.0)])`

Reply:
Thanks, just wanted to make sure, I get it correctly